opsgenie/resource_opsgenie_role.go

```diff
@@ -0,0 +1,220 @@
+package opsgenie
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"log"
+
+	"github.com/opsgenie/opsgenie-go-sdk-v2/custom_user_role"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+var validRights = []string{
+	"who-is-on-call-show-all",
+	"notification-rules-edit",
+	"quiet-hours-edit",
+	"alerts-access-all",
+	"reports-access",
+	"logs-page-access",
+	"maintenance-edit",
+	"contacts-edit",
+	"profile-edit",
+	"login-email-edit",
+	"profile-custom-fields-edit",
+	"configurations-read-only",
+	"configurations-edit",
+	"configurations-delete",
+	"billing-manage",
+	"alert-action",
+	"alert-create",
+	"alert-add-attachment",
+	"alert-delete-attachment",
+	"alert-add-note",
+	"alert-acknowledge",
+	"alert-unacknowledge",
+	"alert-snooze",
+	"alert-escalate",
+	"alert-close",
+	"alert-delete",
+	"alert-take-ownership",
+	"alert-assign-ownership",
+	"alert-add-recipient",
+	"alert-add-team",
+	"alert-edit-tags",
+	"alert-edit-details",
+	"alert-custom-action",
+	"alert-update-priority",
+	"alert-acknowledge-all",
+	"alert-close-all",
+	"incident-create	alert-create",
+	"incident-add-stakeholder",
+	"incident-add-responder",
+	"incident-resolve",
+	"incident-reopen",
+	"mass-notification-create",
+	"service-access",
+}
+
+func resourceOpsGenieCustomUserRole() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOpsGenieCustomUserRoleCreate,
+		Read:   handleNonExistentResource(resourceOpsGenieCustomUserRoleRead),
+		Update: resourceOpsGenieCustomUserRoleUpdate,
+		Delete: resourceOpsGenieCustomUserRoleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"rolename": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"extended_role": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"user", "observer", "stakeholder",
+				}, false),
+			},
+			"granted_rights": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice(validRights, false),
+				},
+				Set: schema.HashString,
+			},
+			"disallowed_rights": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice(validRights, false),
+				},
+				Set: schema.HashString,
+			},
+		},
+	}
+}
+
+func flattenSet(input *schema.Set) []string {
+	output := make([]string, 0)
+
+	if input == nil {
+		return output
+	}
+
+	for _, v := range input.List() {
+		output = append(output, v.(string))
+	}
+	return output
+}
+
+func resourceOpsGenieCustomUserRoleCreate(d *schema.ResourceData, meta interface{}) error {
+
```

Comment:
can you also remove this line in each functions.

Reply:
did not get this, what change should I do in the function header?

Reply:
in order to comply with repo formatting settings please don't break line after the function desc